test/modules/paymentProcessor/abstracts/CrossChainBase_v1.t.sol

```diff
@@ -0,0 +1,177 @@
+        impl = address(new ERC20PaymentClientBaseV2Mock());
+        paymentClient = ERC20PaymentClientBaseV2Mock(Clones.clone(impl));
+        //Adding the payment client is done through a timelock mechanism
+        _orchestrator.initiateAddModuleWithTimelock(address(paymentClient));
+        vm.warp(block.timestamp + _orchestrator.MODULE_UPDATE_TIMELOCK());
+        _orchestrator.executeAddModule(address(paymentClient));
+        //Init payment client
+        paymentClient.init(_orchestrator, _METADATA, bytes(""));
+        paymentClient.setIsAuthorized(address(crossChainBase), true);
+        paymentClient.setToken(_token);
+    }
+    //--------------------------------------------------------------------------
+    //Test: Initialization
+    /*
+    └──  Given the contract is not initialized
+    └── When initializing the contract
+        └── Then it should set the correct orchestrator address */
+
+    function testInit() public override(ModuleTest) {
+        assertEq(address(crossChainBase.orchestrator()), address(_orchestrator));
+    }
+
+    //--------------------------------------------------------------------------
+    //Test: Interface Support
+    /*
+    └── Given the contract is initialized
+    └── When checking for ICrossChainBase_v1 interface support
+        └── Then it should return true
+        └── When checking for an unknown interface
+            └── Then it should return false */
+    function testSupportsInterface() public {
+        // Test for ICrossChainBase_v1 interface support
+        bytes4 interfaceId = type(ICrossChainBase_v1).interfaceId;
+        assertTrue(crossChainBase.supportsInterface(interfaceId));
+    }
+
+    /*  
+    └──  Given the contract is already initialized
+    └── When trying to reinitialize
+        └── Then it should revert with Initializable__InvalidInitialization */
+    function testReinitFails() public override(ModuleTest) {
+        vm.expectRevert(OZErrors.Initializable__InvalidInitialization);
+        crossChainBase.init(_orchestrator, _METADATA, abi.encode(1));
+    }
+
+    /**
+     * @dev Test interface support failure case
+     * └── Given the contract is initialized
+     * └── When checking for an unknown interface
+     *     └── Then it should return false
+     */
+    function testSupportsInterface_failsGivenUnknownInterface() public {
+        bytes4 randomInterfaceId = bytes4(keccak256("random()"));
+        assertFalse(crossChainBase.supportsInterface(randomInterfaceId));
+    }
+
+    //--------------------------------------------------------------------------
+    //Test: executeBridgeTransfer
+
+    /**
+     * @dev Test bridge transfer with empty payment order
+     * └── Given an empty payment order is created
+     * └── When executeBridgeTransfer is called
+     *     └── Then it should return empty bytes
+     */
+    function testExecuteBridgeTransfer_succeedsGivenEmptyPaymentOrder()
+        public
```

Comment:
Should have the name `internal` in it for internal functions. I.e. `testInternalExecuteBridgeTransfer_worksGivenImplemented`

Reply:
Also the function is actually just testing if the function has been implemented (as it should), so the name as well as the Gherkin should be adapted to this